pack.go

```diff
@@ -34,52 +34,102 @@ const (
 type PackOptions struct {
 	// Subject is the subject of the manifest.
 	Subject *ocispec.Descriptor
+
 	// ManifestAnnotations is the annotation map of the manifest.
 	ManifestAnnotations map[string]string
 
 	// PackImageManifest controls whether to pack an image manifest or not.
-	//   - If true, pack an image manifest; artifactType will be used as the
-	// the config descriptor mediaType of the image manifest.
-	//   - If false, pack an artifact manifest.
+	//   - If true, pack an OCI Image Manifest
+	//   - If false, pack an OCI Artifact Manifest
 	// Default: false.
 	PackImageManifest bool
+
+	// PackManifestType controls which type of manifest to pack.
+	// This option is valid only when PackImageManifest is true.
+	// Default: PackManifestTypeImageManifestLegacy.
+	PackManifestType
+
 	// ConfigDescriptor is a pointer to the descriptor of the config blob.
 	// If not nil, artifactType will be implied by the mediaType of the
 	// specified ConfigDescriptor, and ConfigAnnotations will be ignored.
 	// This option is valid only when PackImageManifest is true.
 	ConfigDescriptor *ocispec.Descriptor
+
 	// ConfigAnnotations is the annotation map of the config descriptor.
 	// This option is valid only when PackImageManifest is true
 	// and ConfigDescriptor is nil.
 	ConfigAnnotations map[string]string
 }
 
+// DefaultPackOptions provides the default PackOptions.
+var DefaultPackOptions PackOptions = PackOptions{
+	PackImageManifest: true,
+	PackManifestType:  PackManifestTypeImageManifest,
+}
+
 // Pack packs the given blobs, generates a manifest for the pack,
 // and pushes it to a content storage.
 //
-// When opts.PackImageManifest is true, artifactType will be used as the
-// the config descriptor mediaType of the image manifest.
+//   - If opts.PackImageManifest is true and opts.PackManifestType is
+//     PackManifestTypeImageManifestLegacy (the default value),
+//     artifactType will be used as the the config media type of the image
+//     manifest when opts.ConfigDescriptor is not specified.
+//   - If opts.PackImageManifest is true and opts.PackManifestType is
+//     PackManifestTypeImageManifest, [ErrMissingArtifactType] will be returned
+//     when none of artifactType and opts.ConfigDescriptor is specified.
+//
 // If succeeded, returns a descriptor of the manifest.
 func Pack(ctx context.Context, pusher content.Pusher, artifactType string, blobs []ocispec.Descriptor, opts PackOptions) (ocispec.Descriptor, error) {
-	if opts.PackImageManifest {
+	if !opts.PackImageManifest {
+		return packArtifact(ctx, pusher, artifactType, blobs, opts)
+	}
+
+	switch opts.PackManifestType {
+	case PackManifestTypeImageManifestLegacy:
+		return packImageLegacy(ctx, pusher, artifactType, blobs, opts)
+	case PackManifestTypeImageManifest:
 		return packImage(ctx, pusher, artifactType, blobs, opts)
+	default:
+		return ocispec.Descriptor{}, fmt.Errorf("PackManifestType %v is unsupported: %w", opts.PackManifestType, errdef.ErrUnsupported)
```

Comment:
nit: the word `unsupported` will be printed twice
```suggestion
		return ocispec.Descriptor{}, fmt.Errorf("PackManifestType(%v): %w", opts.PackManifestType, errdef.ErrUnsupported)
```